# HG changeset patch
# User Rik <rik@octave.org>
# Date 1344435801 25200
#      Wed Aug 08 07:23:21 2012 -0700
# Node ID 25200c04bc252803461dd82f1410a8a0c3cd857c
# Parent  067699edef865b3635fd6bb845c0a965188291bc
configure.ac: Require makeinfo to be present before building Octave.

* configure.ac: Add new OCTAVE_PROG_MAKEINFO test.

* m4/acinclude.m4: Add new OCTAVE_PROG_MAKEINFO check for presence of makeinfo.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2053,16 +2053,17 @@ AC_SUBST(INSTALL_SCRIPT)
 
 OCTAVE_PROG_DESKTOP_FILE_INSTALL
 
 OCTAVE_PROG_GNUPLOT
 OCTAVE_PROG_PAGER
 OCTAVE_PROG_GPERF
 
 OCTAVE_PROG_GHOSTSCRIPT
+OCTAVE_PROG_MAKEINFO
 OCTAVE_PROG_TEXI2DVI
 OCTAVE_PROG_TEXI2PDF
 
 ### Need to disable building documentation if gnuplot was not found,
 ### unless it was already disabled previously.
 
 if test -n "$DOCDIR" && test -n "$warn_gnuplot"; then
   DOCDIR=
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -680,16 +680,29 @@ will fail, and saving graphics in some o
 using Octave
 "
 
     OCTAVE_CONFIGURE_WARNING([warn_ghostscript])
   fi
   AC_SUBST(GHOSTSCRIPT)
 ])
 dnl
+dnl Find makeinfo required for reading documentation
+dnl
+dnl OCTAVE_PROG_MAKEINFO
+AC_DEFUN([OCTAVE_PROG_MAKEINFO],
+dnl use MKINFO, not MAKEINFO, for variable name because Automake automatically
+dnl defines a value for MAKEINFO even when it does not exist which will then
+dnl fool the 'test -z' line.
+  [AC_CHECK_PROG(MKINFO, makeinfo, makeinfo, [])
+   if test -z "$MKINFO"; then
+     AC_MSG_ERROR([makeinfo program required for reading documentation])
+   fi
+])
+dnl
 dnl Is texi2dvi installed?
 dnl
 dnl OCTAVE_PROG_TEXI2DVI
 AC_DEFUN([OCTAVE_PROG_TEXI2DVI], [
   AC_CHECK_PROG(TEXI2DVI, texi2dvi, texi2dvi, [])
   if test -z "$TEXI2DVI"; then
     TEXI2DVI='$(top_srcdir)/build-aux/missing texi2dvi'
     warn_texi2dvi="
